### PR TITLE
Fix print jobs incorrectly marked as finished even though some layers were not

### PR DESCRIPTION
**Pull request by jahow**
This was due to an oversight in progress rounding logic. Added a failing test fixed by this change.

Also added unrelated unit tests for the new GeoJSON layer that was added in #42 and #44.

### CODE REVIEW

File: test/unit/printer/job.test.js [hunk 1]
```diff
@@ -57,59 +57,93 @@ describe('job creation', () => {
...
-        sourceLoadErrors: [
-          {
-            url: errorurl,
-          },
-        ],
-      },
+
+  describe('with 100 layers', () => {
+    beforeEach(() => {
+      const layers = new Array(30).fill(0).map((v, i) => ({
```
Comment (LHBruneton-C2C) [inline test/unit/printer/job.test.js:126]: 100 or 30 ? Or am I missing something ?

Comment (jahow) [reply test/unit/printer/job.test.js:126]: good catch! I'll leave it at 100

File: test/unit/printer/layers.test.js [hunk 2]
```diff
@@ -341,6 +342,48 @@ describe('layer creation', () => {
...
+    /** @type {Layer} */
+    const spec = {
+      type: 'GeoJSON',
+      geojson: {
+        type: 'FeatureCollection',
+        name: 'france_3857',
+        crs: {
+          type: 'name',
+          properties: { name: 'urn:ogc:def:crs:EPSG::3857' },
+        },
```
Comment (LHBruneton-C2C) [inline test/unit/printer/layers.test.js:356]: Why put this in the fixture if it's not supported by the api ?

**Review by LHBruneton-C2C — APPROVED** [review]
Looking good, thanks! Only some minor comments.

File: test/unit/printer/layers.test.js (hunk 2 above)
Comment (jahow) [reply test/unit/printer/layers.test.js:356]: good catch

**Pull request merged**